crates/rspack_core/src/options/builtins.rs

```diff
@@ -102,6 +106,97 @@ pub struct Minification {
   pub drop_console: bool,
   pub pure_funcs: Vec<String>,
   pub extract_comments: Option<String>,
+  pub test: Option<MinificationConditions>,
+  pub include: Option<MinificationConditions>,
+  pub exclude: Option<MinificationConditions>,
+}
+pub enum MinificationCondition {
+  String(String),
+  Regexp(RspackRegex),
+}
+
+impl MinificationCondition {
+  #[async_recursion]
+  pub async fn try_match(&self, data: &str) -> rspack_error::Result<bool> {
+    match self {
+      Self::String(s) => Ok(data.starts_with(s)),
+      Self::Regexp(r) => Ok(r.test(data)),
+    }
+  }
+}
+
+impl fmt::Debug for MinificationCondition {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    match self {
+      Self::String(i) => Debug::fmt(&i, f),
+      Self::Regexp(i) => i.fmt(f),
+    }
+  }
+}
+
+impl Hash for MinificationCondition {
```

Comment:
Maybe we don't need Hash for this, compiler is complaining because `Minification` implemented `Hash`, but `Builtins` didn't implement it, maybe you can try to remove the `Hash` on `Minification`, not sure whether it's still useful

Reply:
It is hashed in js_chunk_hash function in rspack_plugin_javascript/plugin/impl_plugin-for_js_plugin.rs, so this cannot be removed